contents/java/JavaConcurrency.md

```diff
@@ -0,0 +1,149 @@
+# Java Concurrency
+
+Authors: Vivek Lakshmanan
+
+* [What is Java Concurrency](#what-isjava-concurrency)
+* [Getting started](#getting-started)
+    * [Understanding Threads](#understanding-threads)
+    * [Creating and Starting Threads](#creating-and-starting-threads)
+    * [Operations on Threads](#operations-on-threads)
+    * [Thread Interference](#thread-interference)
+    * [Executors](#executors)
+* [Disadvantages of Java Concurrency](#disadvantages-of-java-concurrency)
+* [Resources](#resources)
+    * [Further Reading](#further-reading)
+
+## What is Java Concurrency
+_Concurrency_ is the ability to run several programs or several parts of a program **out-of-order**, in an **interleaved fashion**. Simply put, if a program is running concurrently, the program execution will execute one part of the program, pause it, execute another part and repeat. This is different from _Parallelism_ in which parts of the program are executed at the **same time**.
+
+Java Concurrency enables you to use perform tasks using multi-threading in your code and therefore:
```

Comment:
>I think that you should explain the distinction between concurrency and parallelism at the start

This sounds good! As you mentioned above, I haven't separated them properly, so having a clear distinction would improve clarity. 

>but then talk about concurrency in the rest of the chapter like as if concurrency implies parallelism. What do you think?

Since this chapter is about concurrency, I think it's better to replace `take advantage of multi-core processors` with `better resource utilization`. So now, `using both concurrency and parallelism at the same time` is not implied which is more appropriate since that's not what concurrency in Java allows you to do.

On a side note, the term `multi-threading` still falls under concurrency right? Since threads are just `lightweight processes` and don't imply parallelism.

Reply:
Ah yes, that is technically correct re. `multi-threading`. I am also guilty of conflating it with parallelism :sweat_smile: 

> Since this chapter is about concurrency, I think it's better to replace `take advantage of multi-core processors` with `better resource utilization`. So now, `using both concurrency and parallelism at the same time` is not implied which is more appropriate since that's not what concurrency in Java allows you to do.

Yeah, I think that's okay. It's stuff like not letting the core go idle for stuff like waiting on I/O.